src/jbi/services.py

```diff
@@ -13,17 +14,45 @@
 logger = logging.getLogger(__name__)
 
 
+class MethodCounterWrapper:
+    """This class wraps an object and increments a counter everytime
+    the specieds methods are called.
+    """
+
+    counters: Dict[str, Counter] = {}
+
+    def __init__(self, wrapped, methods):
+        self.wrapped = wrapped
+        for method in methods:
+            if method not in self.counters:
+                self.counters[method] = Counter(method, f"Call of {method}()")
```

Comment:
maybe there should be a prefix string here (eg. `prefix="jira."`)

Reply:
or have `prefix` be a param; so this could be reused for the other services?